src/App.tsx

```diff
@@ -1,24 +1,205 @@
+import { Filter } from './types/Filter';
+import { getVisibleTodos } from './helper/getVisibleTodos';
 
-const USER_ID = 0;
+const USER_ID = 6762;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setMessageError] = useState('');
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [title, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inProcessing, setProcessingIDs] = useState<number[]>([0]);
+  const hasError = !!errorMessage || errorMessage !== '';
+  const isDisableInput = !tempTodo;
+
+  const showError = (message: string) => {
+    setMessageError(message);
+    setTimeout(() => setMessageError(''), 3000);
+  };
+
+  const loadingTodos = async () => {
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setTodos(todos);
+    } catch (error) {
+      showError(ErrorNotice.LOADING);
+      setTodos(state => [...state]);
+    }
+  };
+
+  useEffect(() => {
+    loadingTodos();
+  }, []);
+
+  const handleSubmit = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (!title.trim()) {
+        showError(ErrorNotice.TYTLE);
+
+        return;
+      }
+
+      const createTodo = {
+        id: 0,
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setTempTodo(createTodo);
+
+      try {
+        await addTodos(createTodo);
+        setTodos(state => [...state, createTodo]);
+
+        setNewTitle('');
+        setTempTodo(null);
+        loadingTodos();
+      } catch (error) {
+        showError(ErrorNotice.ADD);
+        setTempTodo(null);
+      }
+    }, [title],
+  );
+
+  const handleDelete = useCallback(async (todoId: number) => {
+    try {
+      setProcessingIDs([todoId]);
+      await deleteTodo(todoId);
+      loadingTodos();
+    } catch (error) {
+      showError(ErrorNotice.DELETE);
+    } finally {
+      setProcessingIDs([0]);
+    }
+  }, [todosFromServer]);
+
+  const handleDeleteCompleted = useCallback(
+    async (todos: Todo[]) => {
+      try {
+        setProcessingIDs(todos.map(todo => todo.id));
+        const removeAllComleted = todos.map(todo => (
+          deleteTodo(todo.id)
+        ));
+
+        await Promise.all(removeAllComleted);
+
+        loadingTodos();
+      } catch (error) {
+        showError(ErrorNotice.DELETE);
+      } finally {
+        setProcessingIDs([0]);
+      }
+    }, [inProcessing],
+  );
+
+  const handleComplete = useCallback(
+    async (todoId: number[], data: boolean) => {
+      setProcessingIDs(todoId);
+
+      try {
+        const completingTodos = todoId.map(id => (
+          completeTodo(id, data)
+        ));
+
+        await Promise.all(completingTodos);
+
+        loadingTodos();
+      } catch (error) {
+        showError(ErrorNotice.UPDATE);
+      } finally {
+        setProcessingIDs([0]);
+      }
+    }, [todosFromServer],
```

Comment:
todosFromServer might not be needed as a dependency of handleComplete, consider removing it